src/BBT.StructureTools.Tests/Compare/ComparerWithObjectInheritingFromBaseModelAttributeWithDistinguishedComparerTests.cs

```diff
@@ -0,0 +1,231 @@
+﻿namespace BBT.StructureTools.Tests.Compare
+{
+    using System;
+    using System.Collections.Generic;
+    using BBT.StructureTools;
+    using BBT.StructureTools.Compare;
+    using BBT.StructureTools.Compare.Exclusions;
+    using BBT.StructureTools.Compare.Helper;
+    using BBT.StructureTools.Extension;
+    using BBT.StructureTools.Tests.Compare.Intention;
+    using BBT.StructureTools.Tests.TestTools;
+    using FluentAssertions;
+    using Ninject;
+    using Xunit;
+
+    /// <summary>
+    /// Test for Comparer infrastructure with object attributes.
+    /// </summary>
+    /// <remarks>
+    /// Raison d'Être for this integration test collection is that before a refactoring,
+    /// <see cref="IEqualityComparerHelperRegistration{T}.RegisterAttributeWithDistinguishedComparer{TComparer,TIntention}"/>
+    /// only accepted classes inheriting <see cref="IBaseModel"/> as parameter. The tests here are
+    /// the same as defined in <see cref="ComparerWithObjectAttributeWithDistinguishedComparerIntTests"/>
+    /// but explicitly test using an attribute in the <see cref="TestClass"/> which is inheriting
+    /// from <see cref="IBaseModel"/> and <see cref="BaseModel"/> (called
+    /// <see cref="BaseModelTestAttribute"/>).
+    /// It's paranoid yet justified since we can't be sure about no
+    /// black magic going on within <see cref="BaseModel"/>, and this way we can at least give our
+    /// best to ensure that we're somewhat backwards compatible an don't break anything important
+    /// by accident.
+    /// </remarks>
+    public class ComparerWithObjectInheritingFromBaseModelAttributeWithDistinguishedComparerTests
```

Comment:
Doesn't this test not work in the GitHub scenario? IMHO this test should go into the Core repo, internally at BBT, because it's testing the infrastructures behaviour regarding only internally at BBT available things.

Reply:
Moved to BBT internal code base.